streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -770,7 +771,7 @@ void runOnce() {
         long totalCommitLatency = 0L;
         long totalProcessLatency = 0L;
         long totalPunctuateLatency = 0L;
-        if (state == State.RUNNING) {
+        if (state == State.RUNNING || stateUpdaterEnabled) {
```

Comment:
This seems risky. Can you explain what states this flag could be true?

I am worries about CREATED and ERROR states mostly?

Reply:
I agree with the error states and I will check for those. 
A stream thread in `CREATED` would never reach this code because in `run()` the state is either set to `STARTING` or the stream thread is stopped. I think the state `STARTING` is OK, since with `STARTING` the loop should be a no-op since there would not be any tasks to process.